### ISSUE TITLE
Account util refactoring

### ISSUE THREAD

**Issue by mrpudn** [bamboo-black/ghosts-of-denth#4]
All of the stuff in the `api/accounts/util.js` file should be in either the `model.js` or the `router.js`

**Comment by njgrisafi** [comment]
@mrpudn how about the the mongoose custom validators http://mongoosejs.com/docs/validation.html

**Comment by mrpudn** [comment]
Yup that's exactly what I was thinking. I'm finding that it's very tricky to test object creation with node.js though - hence abstracting out the `create(...)` stuff into a utility module.

I think we need to replace this stuff with mongoose validations, remove the `util.js` module, and test what we can.

**Comment by mrpudn** [comment]
This one will probably get resolved with #2 - I'll go ahead and assign it to me
